File: 268-missing-number/268-missing-number.cpp
class Solution {
public:
    int missingNumber(vector<int>& nums) {
        int n = nums.size();
        
        for( int i = 0; i < n; i++ ) {
            while (nums[i] != i) {
                int d = nums[i];
                if (nums[i] < n)
                    swap( nums[i], nums[d]);
                else
                    break;
            }
        }
            
        for( int i = 0; i < n; i++ ) {
            if(nums[i] == n)
                return i;
        }
         
        return n;
    }
};

File: 268-missing-number/268-missing-number.py
class Solution(object):
    def missingNumber(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        
        for i in range(len(nums)):
            while nums[i] != i:
                d = nums[i]
                if d < len(nums):
                    nums[i], nums[d] = nums[d], nums[i]
                else:
                    break
        
        for i in range(len(nums)):
            if nums[i] == len(nums):
                return i
        
        return len(nums)
    